### DISCUSSION TITLE
CPU peaks - ClamAV module enabled

### DISCUSSION THREAD

**Discussion by vawaver** [nextcloud/all-in-one#2841] [Questions] [answered]
Hello, I would like to share my experience and discuss an issue I encountered with the ClamAV docker.

I noticed that when I installed the full Nextcloud AIO, including the ClamAV module, this module started performing operations every 10 minutes, causing my processor usage to spike up to around 25%, even when the server was not being actively used.

However, when I disabled this module, the problem disappeared.

To address the issue, I attempted an alternative approach. I installed ClamAV using the command 
`sudo apt-get install clamav clamav-daemon -y`. 
Afterwards, I activated the "Antivirus for files" application in the Nextcloud App.

With this configuration, I observed that the processor load remained constant at 1-2%, without any spikes.

Based on this, it can be assumed that enabling ClamAV as a docker container leads to these problems. Therefore, I would appreciate it if you could test and potentially optimize the ClamAV docker configuration.

Hardware - VM on Proxmox -> 8 cores + 15 GB RAM
#### Current channel  stable

![clamav_100](https://github.com/nextcloud/all-in-one/assets/52236863/1055955a-9572-4a43-bb8d-2d0de211341e)
![clamav_peaks](https://github.com/nextcloud/all-in-one/assets/52236863/04c88a90-0f50-4fde-a600-65d8acc85985)
![clamav_aio_settings](https://github.com/nextcloud/all-in-one/assets/52236863/ddba93a0-9b0e-4654-9be3-7b462aeb7eac)

**Comment by szaimen** [comment]
Hi, can you post the nextcloud logs and clamav logs from when this happens?

**Comment by vawaver** [comment]
Meanwhile, I disabled the ClamAV module. 
Once I turn it back on and the process begins, I will attempt to retrieve the logs.
I am aware that, in Nextcloud, this can be done through the Logging tab. Unfortunately, 
I am unfamiliar with accessing the ClamAV logs specifically. I am not an IT professional. 
Could you please advise me on how to proceed?

**Comment by szaimen** [comment]
You can retreive the clamav logs eg. With `sudo docker logs nextcloud-aio-clamav`

**Comment by vawaver** [comment]
I am sending the latest logs.
Let me know whether it is ok for you.

[tony@nxtc:~$ sudo docker logs nextcloud.txt](https://github.com/nextcloud/all-in-one/files/11818718/tony%40nxtc.sudo.docker.logs.nextcloud.txt)
[nextcloud.log](https://github.com/nextcloud/all-in-one/files/11818719/nextcloud.log)

**Comment by vawaver** [comment]
Another Clamav peak - 16:40 CET

log - [clamav-peak-16_40.txt](https://github.com/nextcloud/all-in-one/files/11818931/clamav-peak-16_40.txt)

![CLAMAV_peak_16_40](https://github.com/nextcloud/all-in-one/assets/52236863/d827002f-e335-421d-9c26-1f94fbf120bc)

**Comment by szaimen** [comment]
The issue that is mentioned in the logs is this one: https://github.com/nextcloud/files_antivirus/issues/261. Nothing we can prevent or fix in AIO IIRC - it must be fixed in the files_antivirus app.

---

As a workaround for the problem, can you run `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:app:set av_background_scan --value off` and check if that solves the issue?

---

Also, do you have any suggestion what we can improve in our config here?
https://github.com/nextcloud/all-in-one/blob/5e160b1bfe981daac446febc2f6836cabc524d96/Containers/clamav/clamav.conf#L1-L4
https://github.com/nextcloud/all-in-one/blob/5e160b1bfe981daac446febc2f6836cabc524d96/Containers/nextcloud/entrypoint.sh#L650-L651

**Comment by szaimen** [comment]
BTW, how big is this file for example? /vawaver/files/Documents/Linux/linux-doma/navody/videonavody/old_videotutorials/proxmox-seria/proxmox-02/proxmox-02.dra.zip

**Comment by vawaver** [comment]
> proxmox-02.dra.zip

Filesize of the file is 89 MB. It is the Davinci Resolve video project.

**Comment by szaimen** [comment]
Then I guess it is an issue within the files_antirivus app because the config is allowing up to 100mb, see above.

Can you try thw workaeound from above?

**Comment by vawaver** [comment]
![image](https://github.com/nextcloud/all-in-one/assets/52236863/62fc76b9-6a80-4a53-a21d-10bee5f8c536)

**Comment by szaimen** [comment, marked as answer]
Sorry please try this one:

`sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:app:set files_antivirus av_background_scan --value off`

**Comment by vawaver** [comment]
> Sorry please try this one:
> 
> `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:app:set files_antivirus av_background_scan --value off`

Done. Waiting for some presentable results.
I will let you know.

**Comment by vawaver** [comment]
It seems that the temporary fix works for the moment. I hope ClamAV will be fixed soon.

![image](https://github.com/nextcloud/all-in-one/assets/52236863/c605619e-3629-4e2d-854a-7481fb86ea02)

**Comment by szaimen** [comment]
Good to hear that it solves the problem for you!

**Comment by piotr-sikora-v** [comment]
For me problem was in low memory for Nextcloud server.
Sometimes clamav take too much memory so it's was in restarting loop.
Increasing memory for that VM fixed everything :)
